R/config.R

```diff
@@ -43,194 +46,73 @@ new_config <- function(initial_population, transitions, parameters,
-#'   \code{\link{infect_fun}} for examples for how to write custom functions.
-#'   }
-#'   \item{delay}{
-#'   If TRUE, transition is interpreted as a delay, if FALSE, transition is
-#'   interpreted as a daily probability.
-#'   }
-#'   \item{pred1}{
-#'   Specifies the first predictor to use in a transition function. One of NA,
-#'   a string identical to a value of the "pred" column in the predictors table,
-#'   or a pattern that matches at least one life stage.
-#'   }
-#'   \item{pred2}{
-#'   Specifies the second predictor to use in a transition function. Format like
-#'   pred1.
-#'   }
-#' }
-#'
-#' @param parameters A `tibble` of parameters to use in the transitions
-#' described in the transitions table. Each row corresponds to a parameter
-#' value that may be used in one or more transitions. Parameter values will be
-#' used in transitions where the "from" and "to" fields of the two (parameters
-#' and transitions) tables match.
-#'
-#' \describe{
-#'   \item{from}{
-#'   Used to identify the transitions that a parameter should be used for.
-#'   Format like the "from" column in the transitions table, or a regex pattern
-#'   that matches with one or more life stage strings.
-#'   }
-#'   \item{to}{
-#'   Used to identify the transitions that a parameter should be used for.
-#'   Format like the "to" column in the transitions table, or a regex pattern
-#'   that matches with one or more life stage strings.
-#'   }
-#'   \item{param_name}{
-#'   A string specifying the name of the argument in the function where you want
-#'   to use a parameter.
-#'   }
-#'   \item{host_spp}{
-#'   Optional column, not needed for model configurations that do not dependent
-#'   on host community. For a given row, NA if the parameter value is not
-#'   dependent on the host species. Otherwise, a string specifying the name of
-#'   the host species that the parameter value pertains to.
-#'   }
-#'   \item{param_value}{
-#'   Numeric; the value of the parameter
-#'   }
-#' }
-#'
-#' @param predictors Optionally, a `tibble` of input data to be used as
-#' predictor values in transition functions, for example weather or host
-#' density.
-#'
-#' \describe{
-#'   \item{pred}{String specifying the name of the predictor, e.g. "temp" or
-#'   "host_den}
-#'   \item{pred_subcategory}{This column allows specifying predictors for which
-#'   there are multiple values for a given j_day. Predictor values are sorted by
-#'   this column in the config set up. This ensures that when accessing a
-#'   predictor with multiple values for the same j_day, we get a vector of
-#'   predictor values ordered by this column A typical use for this column is
-#'   to specify the host density of each host species.}
-#'   \item{j_day}{Integer specifying the Julian day, or NA for predictors with
-#'   constant value over time}
-#'   \item{value}{Numeric value of predictor}
-#' }
+#' @param life_cycle A tick \link{\code{life_cycle}}
```

Comment:
Breaking up the `config` into a hierarchy of components that can be built also breaks up the documentation, so `?config` is not so massive

Reply:
Sounds good to me!